cdp4jsonserializer/src/main/java/cdp4jsonserializer/deserializers/ValueArrayDeserializer.java

```diff
@@ -45,20 +47,27 @@ public ValueArrayDeserializer(Class<?> vc) {
 
   @Override
   public ValueArray deserialize(JsonParser jp, DeserializationContext ctxt)
-      throws IOException, JsonProcessingException {
-    String value = jp.getCodec().readValue(jp, String.class);
+      throws IOException {
+    ObjectMapper mapper = (ObjectMapper) jp.getCodec();
+    JsonNode node = mapper.readTree(jp);
+    String value = mapper.writeValueAsString(node);
 
-    int firstBracketIndex = value.indexOf("[");
-    int lastBracketIndex = value.indexOf("]");
+    Pattern patternArray = Pattern.compile("^\\[(.*)\\]$");
```

Comment:
On the C# side we had explicitly to use the SingleLine flag for the correct matching of sequences containing multiline characters.
I think it needs that here as well. Based on the java docs (https://docs.oracle.com/javase/tutorial/essential/regex/pattern.html) i think its equivilant in Java is the DOTALL flag.